test/jdk/java/util/concurrent/CompletableFuture/LostInterrupt.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ThreadLocalRandom;
+import static java.util.concurrent.TimeUnit.DAYS;
+
+/*
+ * @test
+ * @bug 8254350
+ * @run main LostInterrupt
+ * @summary CompletableFuture.get may swallow interrupt status
+ * @key randomness
+ */
+
+// TODO: Rewrite as a CompletableFuture tck test ?
+
+/**
+ * Submits a task that completes immediately, then invokes CompletableFuture.get
+ * with the interrupt status set. CompletableFuture.get should either complete
+ * immediately with the interrupt status set, or else throw InterruptedException
+ * with the interrupt status cleared.
+ */
+public class LostInterrupt {
+    static final int ITERATIONS = 10_000;
+
+    public static void main(String[] args) throws Exception {
+        ThreadLocalRandom rnd = ThreadLocalRandom.current();
```

Comment:
Hi Martin,

Is using a `ThreadLocalRandom` important for the test logic?
I was wondering whether it would be better to use  ( `* @library /test/lib`) `jdk.test.lib.RandomFactory`,
which prints the random seed in the output so that you can reproduce with the same pseudo-random
sequence in case of test failures.

best regards,

-- daniel

Reply:
RandomFactory is probably overkill here as the test just needs to exercise untimed and timed get. So just a random boolean rather than a wide range of random values.

Reply:
OK.